src/fiber.cr

```diff
@@ -10,103 +44,77 @@ end
 class Fiber
   STACK_SIZE = 8 * 1024 * 1024
 
-  @@first_fiber : Fiber? = nil
-  @@last_fiber : Fiber? = nil
+  @@fibers = Thread::LinkedList(Fiber).new
   @@stack_pool = [] of Void*
 
   @stack : Void*
-  @resume_event : Event::Event?
-  @stack_top = uninitialized Void*
+  @resume_event : Crystal::Event?
+  @stack_top = Pointer(Void).null
   protected property stack_top : Void*
   protected property stack_bottom : Void*
-  protected property next_fiber : Fiber?
-  protected property prev_fiber : Fiber?
   property name : String?
 
+  # :nodoc:
+  property next : Fiber?
+
+  # :nodoc:
+  property previous : Fiber?
+
+  # :nodoc:
+  def self.inactive(fiber : Fiber)
+    @@fibers.delete(fiber)
+  end
+
   def initialize(@name : String? = nil, &@proc : ->)
     @stack = Fiber.allocate_stack
     @stack_bottom = @stack + STACK_SIZE
+
     fiber_main = ->(f : Fiber) { f.run }
 
+    # ???
```

Comment:
I think this is because the `@stack_bottom` points to 1 position after the valid region of the whole `@stack`. So is allowing the `stack_ptr` to start from "END-1"

Reply:
Yes, because the stack grows down, so issuing a store to stack_bottom would store off the end of the stack.

Reply:
So I think adding a comment here and apply the fixups in the commits is what is pending to re-review and merge this PR. Right?

Then we can move to add some changes to support MT GC and other required changes for parallelism in others PR.